sherpa/cpp_api/brpc/server.cc

```diff
@@ -0,0 +1,378 @@
+        sherpa::AsrResponse::endpoint_inactive :
+        sherpa::AsrResponse::endpoint_active;
+      // update result
+      if (transcript && strlen(transcript)) {
+        LOG(INFO) << "remote " << cntl->remote_side()
+          << " : pcm size=" << request->audio_data().length()
+          << ", vad_status=" << vad_state
+          << ", transcript = [" << transcript << "]";
+        sherpa::AsrResponse_Path* one_path_ = response->add_nbest();
+        one_path_->set_transcript(transcript);
+        response->set_status(vad_state);
+      }
+      if (sherpa::AsrRequest::stream_end == request->status()) {
+        decoder_->release_stream(remote_id_[remote_name]);
+      }
+    } catch (std::exception const& e) {
+      LOG(WARNING) << e.what();
+    } catch (...) {
+      LOG(WARNING) << "unknow error with StreamingAsrService";
+    }
+  }
+
+ private:
+  // batch decoder
+  std::shared_ptr<AsrDecoder> decoder_;
+  std::unordered_map<std::string, int64_t> remote_id_;
+  std::vector<char> tail_padding_;
+
+  std::mutex mutex_;
+};
+
+static constexpr const char *kUsageMessage = R"(
+Online (streaming) automatic speech recognition RPC server with sherpa.
+
+Usage:
+(1) View help information.
+
+  ./bin/brpc-server --help
+
+(2) Run server
+
+  ./bin/brpc-server \
+    --nn-model=/path/to/cpu_jit.pt \
+    --tokens=/path/to/tokens.txt \
+    --use-gpu=false \
+    --server-port=6006
+)";
+
+int main(int argc, char *argv[]) {
+  // set torch
+  // https://pytorch.org/docs/stable/notes/cpu_threading_torchscript_inference.html
+  torch::set_num_threads(1);
+  torch::set_num_interop_threads(1);
+  torch::NoGradGuard no_grad;
+  torch::jit::getExecutorMode() = false;
+  torch::jit::getProfilingMode() = false;
+  torch::jit::setGraphExecutorOptimize(false);
+
+  // set OnlineAsr option
+  sherpa::ParseOptions po(kUsageMessage);
+  sherpa::OnlineAsrOptions opts;
+  opts.Register(&po);
+  int32_t port = 6006;
+  int32_t idle_timeout_s = 3;
+  po.Register("server-port", &port, "Server port to listen on");
+  po.Register("idle-timeout", &idle_timeout_s,
+      "Connection will be closed in `idle_timeout_s'");
+  po.Read(argc, argv);
+  if (argc <= 1) {
+    po.PrintUsage();
+    exit(EXIT_FAILURE);
+  }
+  opts.decoding_method = "modified_beam_search";
+  opts.endpoint_config.rule1 = sherpa::EndpointRule(false, 2.0, 0.0);
+  opts.endpoint_config.rule3 = sherpa::EndpointRule(true, 1.2, 0.0);
+  opts.endpoint_config.rule3 = sherpa::EndpointRule(false, 0.0, 20);
+  opts.Validate();
+  SHERPA_LOG(INFO) << "decoding method: " << opts.decoding_method;
+
+  // create the server.
+  brpc::Server server;
+  StreamingAsrService asr_service_impl(opts);
+  if (server.AddService(&asr_service_impl,
+        brpc::SERVER_DOESNT_OWN_SERVICE) != 0) {
+    LOG(ERROR) << "Fail to add service";
+    return -1;
+  }
+  brpc::ServerOptions options;
+  options.idle_timeout_sec = idle_timeout_s;
+  if (server.Start(port, &options) != 0) {
```

Comment:
Does it by default listen on all the available interfaces of the current machine?